File: Assignments/Assignment 37/Java Codes/program37_3.java
// 3.Write a program which accept string from user and accept one character. 
// and return index of first occurance of that character.

// Input : "Marvellous Multi OS"
//         M

// Output: 0

// Input:  "Marvellous Multi OS"
//         W

// Output: -1

import java.util.Scanner;

class Test
{
    int FirstChar(String str, char ch)
    {
        int iCnt = 0;
        int iCount = -1;

        char Arr[] = str.toCharArray();

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == ch)
            {
                return iCnt;
            }
        }

        return iCount;
    }
}

class program37_3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int iRet = 0;

        System.out.println("Enter String : ");
        String Arr = sobj.nextLine();

        System.out.println("Enter character : ");
        char cValue = sobj.nextLine().charAt(0);

        Test tobj = new Test();

        iRet = tobj.FirstChar(Arr, cValue);

        System.out.println(iRet);
    }
}

File: Assignments/Assignment 37/C Codes/program37_3.c
// 3.Write a program which accept string from user and accept one character. 
// and return index of first occurance of that character.

// Input : "Marvellous Multi OS"
//         M

// Output: 0

// Input:  "Marvellous Multi OS"
//         W

// Output: -1

#include<stdio.h>

int FirstChar(char *str, char ch)
{
    int iCount = 0;

    while(*str != '\0')
    {
        if(*str == ch)
        {
            return iCount;
            break;
        }

        iCount++;
        str++;
    }
    
    iCount = (-1);
    return iCount;
}

int main()
{
    char Arr[50] = {"\0"};
    char cValue = '\0';
    int iRet = 0;

    printf("Enter String : \n");
    scanf("%[^'\n']", Arr);

    fflush(stdin);

    printf("Enter the character : \n");
    scanf("%c", &cValue);

    iRet = FirstChar(Arr, cValue);

    printf("%d\n", iRet);

    return 0;
}